cEP-0020.md

```diff
@@ -0,0 +1,465 @@
+
+#### Creating the Level Model
+
+A mockup of basic `Level` Django model would look something like:
+
+```python
+class Level(models.Model):
+    name = models.CharField()
+    points_required = model.IntegerField()
+    user = models.ManytoManyField(User)
+    privileges = models.ForeignKey(Privilege)
+
+
+class Privilege(models.Model):
+    name = models.CharField(max_length=500)
+```
+
+And a query to create a level would be:
+
+```python
+>>> privilege = Privilege.objects.create(name='Get assigned to a newcomer issue')
+>>> level1 = Level.obejects.create(name='Newbie',
+                                  points_required=5,
+                                  privileges=privilege
+                                  )
+```
+
+#### Extending the User Model to add points and Levels
+
+Whenever a user perform an acitvity his points will be increased. So, there
+should be some methods to check if the user has achieved a new level:
+
+```python
+class User(models.Model):
+    ...
+    ...
+    points = models.IntegerField()
+    level = models.CharField()
+    ...
+
+    def add_points()
+        a_method_to_update_points_and_level()
+        a_method_for_the_log_of_the_acitities()
+```
+
+#### Creating the Badges model
+
+```python
+class Badge(models.Model):
+    name = models.CharField()
+    description = model.TextField()
+```
+
+The Badge model much like our Level model, defining a range of possible
+awards that a user can receive. Unlike levels, users can have many badges at one
+time. This sets up a many-to-many relationship between badges and users, so we
+need an intermediate model to track which badges a user has earned. We’ll call this
+model Achievements
+
+```python
+class Achievement(models.Model):
+    user_id = ManyToManyField(User)
+    bage_id = ManyToManyField(Badge)
+```
+
+Now, to make awarding badges easier, let’s also add a convenience method to the
+User model:
+
+```python
+def award_badge(name):
+    badge = Badge.objects.filter(name=name)
+    badges.award(name)
+```
+
+### Deployment and User dashbaord
+
+There will be a Django management commands for each of the processes we need
+which will be triggerd on the each build of netlify and show the useful
+updated user metrics and stats on a dashboard page.
+
+The user dashboard will have the following column:
+
+| Name | Important Activity Metrics | Level and Privileges | Badges |
+
+`Name` column will show the names of each newcomer
+
+`Important Activity Metrics` column will have different metrics of each
+activity and the number of times that activity has been done, along with
+the record of the number of questions asked and answers given on coala
+Gitter channels while solving that activity.
+
+`Level and Privileges` column will show the current level of the user
+with and enabled privileges.
+
+`Badges` column will show the badges earned by the user based on their
+activity.
+
+## Conclusion
+
+This project addressed the most of the problems a newcomer/maintainer
```

Comment:
Delete the first "the" and make "addressed" present tense:

"This project **addresses** ~~the~~ most of the problems..."